introspection-engine/introspection-engine-tests/tests/re_introspection/relation_mode/mssql.rs

```diff
@@ -0,0 +1,402 @@
+// @@map
+mod at_at_map {
+    use indoc::indoc;
+    use introspection_engine_tests::test_api::*;
+
+    // referentialIntegrity = "prisma" with @@map loses track of the relation policy ("prisma") but preserves relations.
+    #[test_connector(tags(Mssql))]
+    async fn referential_integrity_prisma_at_map_map(api: &TestApi) -> TestResult {
+        let init = formatdoc! {r#"
+            CREATE TABLE [dbo].[foo_table] (
+                [id] INT NOT NULL,
+                [bar_id] INT NOT NULL,
+                CONSTRAINT [foo_table_pkey] PRIMARY KEY CLUSTERED ([id]),
+                CONSTRAINT [foo_table_bar_id_key] UNIQUE NONCLUSTERED ([bar_id])
+            );
+            
+            CREATE TABLE [dbo].[bar_table] (
+                [id] INT NOT NULL,
+                CONSTRAINT [bar_table_pkey] PRIMARY KEY CLUSTERED ([id])
+            );
+        "#};
+
+        api.raw_cmd(&init).await;
+
+        let input = indoc! {r#"
+            generator client {
+                provider        = "prisma-client-js"
+                previewFeatures = ["referentialIntegrity"]
+            }
+
+            datasource db {
+                provider             = "sqlserver"
+                url                  = env("TEST_DATABASE_URL")
+                referentialIntegrity = "prisma"
+            }
+
+            model Foo {
+                id     Int @id
+                bar    Bar @relation(fields: [bar_id], references: [id])
+                bar_id Int @unique
+
+                @@map("foo_table")
+            }
+
+            model Bar {
+                id  Int  @id
+                foo Foo?
+
+                @@map("bar_table")
+            }
+        "#};
+
+        let expected = expect![[r#"
+            generator client {
+              provider        = "prisma-client-js"
+              previewFeatures = ["referentialIntegrity"]
+            }
+
+            datasource db {
+              provider = "sqlserver"
+              url      = env("TEST_DATABASE_URL")
+            }
+
+            model Foo {
+              id     Int @id
+              bar    Bar @relation(fields: [bar_id], references: [id])
+              bar_id Int @unique
+
+              @@map("foo_table")
+            }
+
+            model Bar {
+              id  Int  @id
+              foo Foo?
+
+              @@map("bar_table")
+            }
+        "#]];
+
+        let result = api.re_introspect_config(input).await?;
+        expected.assert_eq(&result);
+
+        Ok(())
+    }
+
+    // referentialIntegrity = "foreignKeys" with @@map loses track of the relation policy ("foreignKeys"), but preserves @relations, which are moved to the bottom.
```

Comment:
I'm wondering whether we should be testing `relationIntegrity = "foreignKeys"` more than once, since that's the default everywhere. Let's keep these tests since they're already there.

Reply:
How do you mean, "more than once"? I'm not sure I'm following

Reply:
once would be on just sqlite, or just postgres, for example